### ISSUE TITLE
onAnyFilterChanged is not properly wired to be invoked by FilterManager

### ISSUE THREAD

**Issue by namnn96** [ag-grid/ag-grid#5492]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**

`onAnyFilterChanged` method from custom filter component is not invoked by `FilterManager` due to `onAnyFilterChanged` not being a part of meta data for filter (at https://github.com/ag-grid/ag-grid/blob/b2d442546bbc90edb3f64c94b21f3a9dfd4ba317/community-modules/core/src/ts/components/framework/componentMetadataProvider.ts#L91)
-> when trying to access the proxy method of the wrapper filter component, the method doesn't exist

**Expected behavior**

`onAnyFilterChanged` method is added to `optionalMethodList` of both `filter` and `filterComponent`, so that `BaseComponentWrapper` can loop and create a proxy at https://github.com/ag-grid/ag-grid/blob/b2d442546bbc90edb3f64c94b21f3a9dfd4ba317/community-modules/core/src/ts/components/framework/frameworkComponentWrapper.ts#L35.

This is for `FilterManager` to invoke at https://github.com/ag-grid/ag-grid/blob/b2d442546bbc90edb3f64c94b21f3a9dfd4ba317/community-modules/core/src/ts/filter/filterManager.ts#L334.

**Please tell us about your environment:**

* **AG Grid version:** 28.0.2

* **Browser:**

Any browser

* **Language:** [all | TypeScript 4.7.4 | ES6/7 | ES5]

**Comment by StephenCooper** [comment]
Hi @namnn96, 

Do you have a code example where this causes an issue for you? Would help us evaluate this ticket.

**Comment by erados** [comment]
I have same problem and here is code example, https://plnkr.co/edit/sYk4mqtYWUfy3PNh?open=personFilter.jsx&preview

**Comment by StephenCooper** [comment]
Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-7266 [Filters] onAnyFilterChanged method not called for Frameworks**

See whether this item will be part of our next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-XXXX) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!
